### ISSUE TITLE
Bridged resources are not replaced when a nested property is updated on a collection with ForceNew

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-terraform-bridge#2753]
### What happened?

Under both TF and Pulumi, a resource is not replaced when updating a property of an element in a ForceNew collection.

### Example

```
[ 
  {nested: val2},
  {nested: val3},
  {nested: val1},
]
```

to

```
[ 
  {nested: val2},
  {nested: val4},
  {nested: val1},
]
```

Does not trigger a replacement of the resource, even if the containing collection is ForceNew.

AWS example by @t0yv0 here: https://github.com/pulumi/pulumi-terraform-bridge/issues/2726#issuecomment-2546152296

```hcl
data "aws_ami" "ubuntu" {
  most_recent = true

  filter {
    name   = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  owners = ["099720109477"] # Canonical
}


resource "aws_instance" "example" {
  ami           = data.aws_ami.ubuntu.id
  instance_type = "t2.micro"

  root_block_device {
    volume_size = 8
  }

  ephemeral_block_device {
    virtual_name = "ephemeral0"
    device_name  = "/dev/sde"
  }

  ephemeral_block_device {
    virtual_name = "ephemeral3" # tweak this
    device_name  = "/dev/sdf"
  }
}
```

Related TF discussions https://github.com/hashicorp/terraform-plugin-sdk/issues/1314 and https://github.com/hashicorp/terraform/issues/34691

Note also that the SDKv2 Schema validation is slightly off in a related way: https://github.com/pulumi/pulumi-terraform-bridge/pull/2723#issuecomment-2541518646

### Output of `pulumi about`

.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by VenelinMartinov** [comment]
Closing as intended.

In TF collections of plain elements are replaced when the number of elements changes or one of the element changes.

Collections of objects are replaced when the number of elements changes. Changing a property of an element does not trigger the collection to be replaced.

**Comment by t0yv0** [comment]
> Collections of objects are replaced when the number of elements changes

Couple of additional information here. 

1. I've confirmed that current Pulumi provider matches TF behavior on the `ephemeral_block_device` example.

2. The same instance example can be used to try lists of MaxItems=1 objects, for example:

```
  private_dns_name_options {
     enable_resource_name_dns_a_record = true
  }
```

I have confirmed that both Pulumi and TF generate update plans when tweaking these. 

So then, it would indeed appear that collection-level ForceNew on `List[{"x": X}]` and `Set[{"x": X}]` does not trigger replacements. when an element's property changes, but it would trigger replacements for simple types like `List[string]`. 

Indeed this appears to be by-design we need to match.
